build-mbl/build.sh

```diff
@@ -185,66 +185,87 @@ define_conf()
   rm -f "$path.new"
 }
 
-## Setup the yocto source archiver
-## PATH: The path to local.conf
-## ARCHIVER: The bitbake archiver source mode to install.  One of
-##           original patched configured.
-##
-## Refer to the bitbake documentation for details of the archiver:
-## https://github.com/openembedded/openembedded-core/blob/master/meta/classes/archiver.bbclass
-##
-
-setup_archiver()
+inject_custom_data()
 {
   local path="$1"
-  local archiver="$2"
-  local name="build-mbl-archiver"
+  local custom_data="$2"
+  local name="$3"
 
-  # We need to modify the local.conf file to inherit the archiver
-  # class then configure the archiver behaviour.
+  # We need to modify the local.conf file to inject custom string the user
+  # specifies.
 
-  # The local.conf modification should be indempotent and handle the
+  # The modification should be indempotent and handle the
   # situation that consecutive builds (and hence consecutive calls to
-  # this setup function) change the archiver configuration, or indeed,
-  # remove the configuration.
+  # this function) change the custom data, or indeed, remove it
   #
   # We achieve this by putting start and end markers around the
   # configuration we inject.  We then implement a 3 stage processes:
   # 1) Remove any existing marked configuration
   # 2) Add the new required configuration, if any.
-  # 3) Atomically update the local.conf file with the new file if
+  # 3) Atomically update the file file with the new file if
   #    anything changed.
   #
   # The atomic update ensures that if we fail, or are interrupted, we
-  # never leave a partially updated local.conf.  Atomic update is not
+  # never leave a partially updated the file  Atomic update is not
   # because another process might race us!
 
   # If we previously aborted we may have left a working temporary file
   # littering the work area.
   rm -f "$path.new"
 
   # Remove any existing marked configuration.
-  awk '/^# start '"$name"'$/ { drop = 1; next }
-       /^# stop '"$name"'$/  { drop = 0; next }
-       drop                  { next }
-                             { print }' "$path" > "$path.new"
+  sed '/^# start '"$name"'$/,/^# stop '"$name"'$/d' "$path" > "$path.new"
 
   # Add the new required configuration, if any.
+  if [ -n "${custom_data}" ]; then
+    printf "# start %s\n%b\n# stop %s\n" "$name" "$custom_data" "$name" >> "$path.new"
+  fi
+
+  # If anything changed then atomically update the file
+  if ! cmp -s "$path" "$path.new"; then
```

Comment:
I don't see the point in the `cmp`. We could just simplify by doing the `mv` unconditionally.

Reply:
Addressed here: d4f4b996b5f332933ef3e1dbfdc37cd5f5f5677f